File: templates/web-spring-dbutil/src/main/java/${packagePath}/util/ServiceException.java
package ${packageName}.util;

/**
 * Service层公用的Exception.
 * @author ${author}
 * @since ${now?string("yyyy-MM-dd zzzz")}
 */
public class ServiceException extends Exception {

	private static final long serialVersionUID = 3583566093089790852L;

	public ServiceException() {
		super();
	}

	public ServiceException(String message) {
		super(message);
	}

	public ServiceException(Throwable cause) {
		super(cause);
	}

	public ServiceException(String message, Throwable cause) {
		super(message, cause);
	}
}

File: templates/web-spring-boot-mybatis/src/main/kotlin$if{lang=kotlin}/${packagePath}/exception/ServiceException.kt
package ${packageName}.exception

import org.springframework.http.HttpStatus

/**
 * 自定义异常 - 服务器内部错误
 * @author ${author}
 * @since ${now?string("yyyy-MM-dd zzzz")}
 */
class ServiceException : CodeException {

    constructor() {
        this.code = HttpStatus.INTERNAL_SERVER_ERROR.value()
    }

    constructor(s: String) : super(s) {
        this.code = HttpStatus.INTERNAL_SERVER_ERROR.value()
    }

    constructor(code: Int, s: String) : super(s) {
        this.code = code
    }

    constructor(s: String, throwable: Throwable) : super(s, throwable) {
        this.code = HttpStatus.INTERNAL_SERVER_ERROR.value()
    }

    constructor(throwable: Throwable) : super(throwable) {
        this.code = HttpStatus.INTERNAL_SERVER_ERROR.value()
    }

}